### PR TITLE
Add separate Querier for Ruler

### PR DESCRIPTION
**Pull request by saswatamcode**
Pending ADR.

This PR,

- Removes defunct stateless rulers
- Adds new separate Querier for Rulers to query data from

The architectural changes can be represented as 

Before,
image

After,
image

To deploy,
- Bump metrics template ref and add new OpenShift params
- Bump ServiceMonitor refs
- Bump dashboard ref
- Copy over modified rules

### CODE REVIEW

File: services/observatorium-metrics-template.jsonnet [hunk 1]
```diff
@@ -68,6 +68,15 @@ local obs = import 'observatorium.libsonnet';
...
     { name: 'THANOS_QUERIER_MAX_CONCURRENT', value: '20' },
+    { name: 'THANOS_RULER_QUERIER_CPU_LIMIT', value: '1' },
+    { name: 'THANOS_RULER_QUERIER_CPU_REQUEST', value: '100m' },
+    { name: 'THANOS_RULER_QUERIER_LOG_LEVEL', value: 'info' },
+    { name: 'THANOS_RULER_QUERIER_MEMORY_LIMIT', value: '1Gi' },
+    { name: 'THANOS_RULER_QUERIER_MEMORY_REQUEST', value: '256Mi' },
+    { name: 'THANOS_RULER_QUERIER_REPLICAS', value: '3' },
+    { name: 'THANOS_RULER_QUERIER_PROXY_STRATEGY', value: 'eager' },
+    { name: 'THANOS_RULER_QUERIER_ENGINE', value: 'prometheus' },
+    { name: 'THANOS_RULER_QUERIER_MAX_CONCURRENT', value: '20' },
```
Comment (douglascamata) [inline services/observatorium-metrics-template.jsonnet:79]: Can we move all these to the block of configuration variables for Thanos Ruler below? The list of variables seems to be sorted alphabetically and I believe it's great for organization purposes, making it easy to find related configuration.

**Review by douglascamata — CHANGES_REQUESTED** [review]
Just one change suggested.

File: services/observatorium-metrics-template.jsonnet (hunk 1 above)
Comment (saswatamcode) [reply services/observatorium-metrics-template.jsonnet:79]: Will do!

**Comment by saswatamcode** [comment]
Yeah fixed in #434 , let me rebase

**Review by douglascamata — APPROVED** [review]
LGTM

**Comment by saswatamcode** [comment]
Deployment in test cluster looks ok, but there is no data to test. Let's merge and deploy to stage, then iterate.

**Pull request merged**